pcdsdevices/ccm.py

```diff
@@ -26,20 +27,77 @@
 default_gd = 231.303
 
 
-class CCMMotor(PVPositionerPC, FltMvInterface):
+class CCMMotor(PVPositioner, FltMvInterface):
     """
     Goofy records used in the CCM.
-
-    TODO: switch to PVPositioner subclass and override code that prevents
-    loading, and make the wait for done just compare the values.
     """
 
     setpoint = Cpt(EpicsSignal, ":POSITIONSET", auto_monitor=True)
     readback = Cpt(EpicsSignalRO, ":POSITIONGET", auto_monitor=True,
                    kind='hinted')
+    done = Cpt(InternalSignal, value=0)
+    done_value = 1
 
     limits = None
 
+    def __init__(self, prefix, *, name, **kwargs):
+        self._last_readback = None
+        self._last_setpoint = None
+        super().__init__(prefix, name=name, **kwargs)
+
+    @readback.sub_value
+    def _update_readback(self, *args, value, **kwargs):
+        """Callback to cache the readback and update done state."""
+        self._last_readback = value
+        self._update_done()
+
+    @setpoint.sub_value
+    def _update_setpoint(self, *args, value, **kwargs):
+        """Callback to cache the setpoint and update done state."""
+        self._last_setpoint = value
+        # Always set done to False when a move is requested
+        # This means we always get a rising edge when finished moving
+        # Even if the move distance is under our done moving tolerance
+        self.done.put(0, force=True)
+        self._update_done()
+
+    def _update_done(self):
+        """
+        Update our status to done if we are within the tolerance.
+
+        This tolerance of 3e-4 was copied as-is from old xcs python code.
+        """
+        if None not in (self._last_readback, self._last_setpoint):
```

Comment:
I like this implementation. One day, maybe we could pull it out into a new class `PVPositionerComparator` (only requiring either `atol` or a callable that takes in setpoint/readback or something). 🤔

Reply:
I agree, I had the same thought. Not sure about how to optimize the general case API.